backend/dataall/tasks/data_sharing/share_managers/s3_share_manager.py

```diff
@@ -397,7 +398,7 @@ def delete_dataset_bucket_key_policy(
                 json.dumps(policy)
             )
 
-    def handle_share_failure(self, error: Exception) -> bool:
+    def handle_share_failure(self, error: Exception) -> None:
```

Comment:
Please update the function description, maybe also consider renaming the function as it only logs now.